core/bentley/src/Compare.ts

```diff
@@ -92,3 +92,47 @@ export function areEqualPossiblyUndefined<T, U>(t: T | undefined, u: U | undefin
   else
     return areEqual(t, u);
 }
+
+export type PrimitiveType = number | string | boolean;
+export function comparePrimitives(lhs: PrimitiveType, rhs: PrimitiveType): number {
+  // Make sure the types are the same
+  if (typeof lhs !== typeof rhs) {
+    return 1;
+  }
+
+  let cmp = 0;
+  // Compare actual values
+  switch (typeof lhs) {
+    case "number":
+      cmp = compareNumbers(lhs, rhs as number);
+      break;
+    case "string":
+      cmp = compareStrings(lhs, rhs as string);
+      break;
+    case "boolean":
+      cmp = compareBooleans(lhs, rhs as boolean);
+      break;
+  }
+  return cmp;
+}
+
+export function comparePrimitiveArrays (lhs?: Array<number|string|boolean>, rhs?: Array<number|string|boolean> ) {
+  if (undefined === lhs)
+    return undefined === rhs ? 0 : -1;
+  else if (undefined === rhs)
+    return 1;
+  else if (lhs.length === 0 && rhs.length === 0) {
+    return 0;
+  } else if (lhs.length !== rhs.length) {
+    return Math.abs(lhs.length - rhs.length);
```

Comment:
Why absolute value? It breaks the [strict weak constraint].

Reply:
right, I will remove it.